ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -920,9 +973,12 @@ Matmul create_matmul_struct(
     auto arch = input_tensor_a.device()->arch();
     const bool has_user_grid = parameters.user_core_coord.has_value();
     const bool has_program_config = parameters.program_config.has_value();
-    const auto increase_fidelity = !has_program_config && !has_user_grid;
+    bool are_inputs_low_precision_df =
+            ((input_tensor_a.get_dtype() == DataType::BFLOAT8_B || input_tensor_a.get_dtype() == DataType::BFLOAT4_B) &&
+            (input_tensor_b.get_dtype() == DataType::BFLOAT8_B || input_tensor_b.get_dtype() == DataType::BFLOAT4_B));
+    const auto increase_fidelity = !has_program_config && !has_user_grid && !are_inputs_low_precision_df;
     auto math_fidelity = increase_fidelity ? MathFidelity::HiFi2 : MathFidelity::LoFi;
-    auto kernel_config_val = init_device_compute_kernel_config(arch, parameters.compute_kernel_config, math_fidelity);
+    auto kernel_config_val = init_device_compute_kernel_config(arch, parameters.compute_kernel_config, math_fidelity, false /*default_approx_mode*/, false /*default_fp32_acc*/, true /*default_l1_acc*/);
```

Comment:
thanks for adding comments to unnamed values passed! 
nit: `false /*default_approx_mode*/` -> `/* default_approx_mode */ false`

Reply:
The style has changed over time.
The current style seems to be 
https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments
`/*default_approx_mode=*/false`

Reply:
chnaged to latest